dogfooding/lib/core/repos/token_service.dart

```diff
@@ -66,13 +74,27 @@ class TokenResponse {
 }
 
 class TokenService {
-  const TokenService();
+  final CustomEnvironmentLoader? customEnvironmentLoader;
+
+  const TokenService({this.customEnvironmentLoader});
 
   Future<TokenResponse> loadToken({
     required String userId,
     required Environment environment,
     Duration? expiresIn,
   }) async {
+    if (environment == Environment.custom) {
+      if (customEnvironmentLoader == null) {
+        throw Exception(
+            'Custom environment is not configured. Provide customEnvironmentLoader to AppInjector.init method to use it.');
+      }
+
+      return TokenResponse(
+        customEnvironmentLoader!.tokenLoader(userId),
+        customEnvironmentLoader!.apiKey,
+      );
```

Comment:
If you make tokenLoader async you need
```suggestion
      final userToken = await customEnvironmentLoader!.tokenLoader(userId);

      return TokenResponse(
        userToken,
        customEnvironmentLoader!.apiKey,
      );
```